src/Nevermind/Nevermind.Evm/TransactionStore.cs

```diff
@@ -0,0 +1,46 @@
+using System.Collections.Generic;
+using Nevermind.Core;
+using Nevermind.Core.Crypto;
+
+namespace Nevermind.Evm
+{
+    public class TransactionStore : ITransactionStore
+    {
+        private readonly Dictionary<Keccak, Transaction> _transactions = new Dictionary<Keccak, Transaction>();
+        private readonly Dictionary<Keccak, TransactionReceipt> _transactionRecepits = new Dictionary<Keccak, TransactionReceipt>();
+        private readonly HashSet<Keccak> _processedTransations = new HashSet<Keccak>();
+        private readonly Dictionary<Keccak, Keccak> _blockHashes = new Dictionary<Keccak, Keccak>();
+
+        public void AddTransaction(Transaction transaction)
+        {
+            _transactions[transaction.Hash] = transaction;
+        }
+
+        public void AddTransactionReceipt(Keccak transactionHash, TransactionReceipt transactionReceipt, Keccak blockhash)
+        {
```

Comment:
let me know how it discriminates between processed and failed or succeeded and also if it matters whether it was included in the main chain